src/api/spec/jobs/status_history_rescaler_job_spec.rb

```diff
@@ -3,32 +3,52 @@
 RSpec.describe StatusHistoryRescalerJob, type: :job do
   include ActiveJob::TestHelper
 
+  let(:status_histories) { StatusHistory.where(key: 'busy_x86_64').order(:time) }
+
   describe '#rescale' do
-    before do
-      Timecop.freeze(2010, 7, 12)
-
-      now = Time.now.to_i - 2.days
-      StatusHistory.transaction do
-        1000.times do |i|
-          StatusHistory.create(time: now + i, key: 'idle_x86_64', value: i)
-        end
+    context 'newer than 2 days events' do
+      before do
+        5.times { |i| StatusHistory.create(time: (Time.now.utc - i.seconds).to_i, key: 'busy_x86_64', value: 10 * i) }
       end
 
-      StatusHistory.create(time: Time.now.to_i, key: 'busy_x86_64', value: 100)
+      subject! { StatusHistoryRescalerJob.perform_now }
+
+      it 'keeps the StatusHistory with key = busy_x86_64' do
+        expect(status_histories.count).to eq(5)
+        expect(status_histories.first.value).to eq(40)
+        expect(status_histories.last.value).to eq(0.0)
+      end
     end
 
-    after do
-      Timecop.return
+    context 'newer than 9 days' do
+      before do
+        2.times { |i| StatusHistory.create(time: 9.days.ago + i.minutes, key: 'busy_x86_64', value: 10 * i) }
+        2.times { |i| StatusHistory.create(time: 3.hours.ago + i.minutes, key: 'busy_x86_64', value: 10 * i) }
+      end
+
+      subject! { StatusHistoryRescalerJob.perform_now }
+
+      it 'reduces the records older than 7 days to 1' do
+        expect(StatusHistory.count).to eq(3)
+      end
     end
 
-    subject! { StatusHistoryRescalerJob.new.perform }
+    context 'older than one month' do
+      before do
+        2.times { |i| StatusHistory.create(time: 9.months.ago + i.minutes, key: 'busy_x86_64', value: 10 * i) }
+        2.times { |i| StatusHistory.create(time: 3.months.ago + i.minutes, key: 'busy_x86_64', value: 10 * i) }
+        2.times { |i| StatusHistory.create(time: 3.hours.ago + i.minutes, key: 'busy_x86_64', value: 10 * i) }
+      end
 
-    it { expect(StatusHistory.count).to eq(2) }
+      subject! { StatusHistoryRescalerJob.perform_now }
+
+      it 'reduces the records older than 1 month to 1' do
+        expect(StatusHistory.count).to eq(4)
```

Comment:
here it get's resampled as expected:
all records older than one month are going to be reduced.
 
We have 2 slots over 1 month with 2 records each. Total 4 records that get reduced to 2 records (Records within the same time slot, get merged together in one new record). So in the end we have 4 records.  2 reduced and 2 intact (because they were created in the last 24 hours)